molt/src/commands.rs

```diff
@@ -1121,6 +1121,31 @@ pub fn cmd_string_equal(_interp: &mut Interp, _: ContextID, argv: &[Value]) -> M
     }
 }
 
+/// string first *needleString* *haystackString* ?*startIndex*?
+pub fn cmd_string_first(_interp: &mut Interp, _: ContextID, argv: &[Value]) -> MoltResult {
+    check_args(2, argv, 4, 5, "needleString haystackString ?startIndex?")?;
+
+    let needle = argv[2].as_str();
+    let haystack = argv[3].as_str();
+
+    let start = if argv.len() == 5 {
+        argv[4].as_int()?
+    } else {
+        0
+    };
+
+    let pos = if start < 0 || start as usize >= haystack.len() {
```

Comment:
The length of a string is `haystack.chars().count()`.  Otherwise you're getting the number of UTF-8 bytes, which is different.  Apparently using `chars().count()` isn't flawless, either, as sometimes you get something that's only part of a printed character because the printed character is made up of multiple graphemes.  (I think grapheme is the right word.)

Reply:
It would be nice and useful to have a command for splitting a string into graphemes.  I don't think graphemes should be the unit for indexing a string because other Unicode Tcl implementations work with code points.